pkg/gateway/watcher.go

```diff
@@ -0,0 +1,26 @@
+package gateway
+
+type InternalConditionWatcher interface {
+	WatchEvents()
```

Comment:
may want to have this interface implement contracts for something like `RegisterEvent(eventType, funcs ...)` & `HandleEvent(eventType)`?

Reply:
@alexandreLamarre Added, let me know if I missed anything.

Reply:
LGTM, should probably add unit tests for these